packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -37,110 +37,91 @@ type RouterAction =
 	| { type: 'gotoparent'; options?: NavigateToParentOptions };
 
 type RouterState = {
+	initialPath: string;
 	screens: Screen[];
-	locationHistory: NavigatorLocation[];
+	currentLocation?: NavigatorLocation;
```

Comment:
Is this type precise now? Seems like we're never setting `isInitial` to `currentLocation`, and we're only setting it manually when computing the context. Should we clean up?

Reply:
The type was precise, but you're right about the fact that we could be setting `isInitial` inside the reducer (instead of doing so in the context value). Done in https://github.com/WordPress/gutenberg/pull/64675/commits/29e0fb2eb74d018b8f52233c8f53a6b803ebdecb